newIDE/app/src/ObjectsRendering/Renderers/CustomObjectLayoutingModel.js

```diff
@@ -85,43 +85,65 @@ const layoutFields = [
   'IsScaledProportionally',
 ];
 
+export const getProportionalPositionX = (positionName: string): ?number => {
+  const horizontalPositionName = (positionName.includes('-')
+    ? positionName.split('-')[1]
+    : positionName
+  ).toLowerCase();
+  return horizontalPositionName === 'left'
+    ? 0
+    : horizontalPositionName === 'right'
+    ? 1
+    : horizontalPositionName === 'center'
+    ? 0.5
+    : null;
+};
+
+export const getProportionalPositionY = (positionName: string): ?number => {
+  const verticalPositionName = (positionName.includes('-')
+    ? positionName.split('-')[0]
+    : positionName
+  ).toLowerCase();
+  return verticalPositionName === 'top'
+    ? 0
+    : verticalPositionName === 'bottom'
+    ? 1
+    : verticalPositionName === 'center'
+    ? 0.5
+    : null;
+};
+
 const getHorizontalAnchorValue = (
   anchorName: string,
-  properties: ?gdMapStringPropertyDescriptor
+  properties: gdMapStringPropertyDescriptor
 ): ?number => {
   const horizontalAnchorName = (anchorName.includes('-')
     ? anchorName.split('-')[1]
     : anchorName
   ).toLowerCase();
-  return horizontalAnchorName === 'left'
-    ? 0
-    : horizontalAnchorName === 'right'
-    ? 1
-    : horizontalAnchorName === 'center'
-    ? 0.5
+  const proportionalX = getProportionalPositionX(horizontalAnchorName);
+  return proportionalX != null
```

Comment:
use a proper `!==` ?

Reply:
According to this documentation, it allows to check for null or undefined. It seems safer even if it doesn't change anything in this case and Flow would detect the issue.
https://flow.org/en/docs/types/maybe/

Reply:
I wonder if we should still manually check for undefined for clarity?